userapi/src/main/java/fk/prof/userapi/Configuration.java

```diff
@@ -0,0 +1,107 @@
+package fk.prof.userapi;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.json.JsonObject;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.Map;
+
+/**
+ * Created by gaurav.ashok on 15/05/17.
+ */
+public class Configuration {
+
+  @JsonProperty("profile.retention.duration.min")
+  public Integer profileRetentionDurationMin = 30;
```

Comment:
In my opinion, keeping something as crucial as configuration mutable is very risky. A wrong line of code someplace, somewhere can cause headache. Usually, we are used to treating config file as a source of truth, with the assumption that whatever we see in the config file is the value being used everywhere.
Let's maintain that invariant. I understand that adding getter is a little more cruft (avoiding setters is the whole point), but this is just one class and I don't foresee config changes all the time. Maybe once with every major feature being shipped.